CHANGELOG.md

```diff
@@ -5,6 +5,96 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](https://semver.org/).
 
+## [v0.38.0] - 2023-06-27
+
+### Added
+* feat(raw/http_util): Implement mixed multipart parser by @Xuanwo in https://github.com/apache/incubator-opendal/pull/2430
+* feat(services/gcs): Add batch delete support by @wcy-fdu in https://github.com/apache/incubator-opendal/pull/2142
+* feat(core): Add Write::sink API by @Xuanwo in https://github.com/apache/incubator-opendal/pull/2440
```

Comment:
Add `Write::sink` API is a breaking change to users who depends on raw API. We should update `upgrade.md`.

Reply:
https://github.com/apache/incubator-opendal/pull/2555/commits/d729d7ebf8713600f6dd7377286d613b9ea6d0bf